### PR TITLE
en, zh: update GCP name to Google Cloud

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added, or deleted? (Required)

- Update outdated term "GCP" to "Google Cloud" because "[Google Cloud Platform is now called Google Cloud](https://cloud.google.com/blog/topics/developers-practitioners/introducing-new-homepage-google-cloud)"
- Update related descriptions

Note: I searched "GCP" and "Google Cloud Platform" in the whole main branch of this repo.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.5 (TiDB Operator 1.5 versions)
- [ ] v1.4 (TiDB Operator 1.4 versions)
- [ ] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):  https://github.com/pingcap/docs-cn/pull/14041

### CODE REVIEW

File: zh/replace-nodes-for-cloud-disk.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ summary: 介绍如何为使用云存储的 TiDB 集群更换节点。
 
 > **注意：**
 >
-> 其它公有云环境请参考[GCP GKE](deploy-on-gcp-gke.md)、[Azure AKS](deploy-on-azure-aks.md)、[阿里云 ACK](deploy-on-alibaba-cloud.md) 操作节点组。
+> 其它公有云环境请参考 [Google Cloud GKE](deploy-on-gcp-gke.md)、[Azure AKS](deploy-on-azure-aks.md)、[阿里云 ACK](deploy-on-alibaba-cloud.md) 操作节点组。
```
Comment (Oreoxmt) [inline zh/replace-nodes-for-cloud-disk.md:14]:
```suggestion
> 其它公有云环境请参考 [Google Cloud GKE](deploy-on-gcp-gke.md)、[Azure AKS](deploy-on-azure-aks.md)或[阿里云 ACK](deploy-on-alibaba-cloud.md) 操作节点组。
```

File: zh/tidb-operator-overview.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ TiDB Operator 提供了多种方式来部署 Kubernetes 上的 TiDB 集群：
     - 在公有云上部署生产可用的 TiDB 集群并进行后续的运维管理；
 
         - [Deploy TiDB on AWS EKS](deploy-on-aws-eks.md)
```
Comment (Oreoxmt) [inline zh/tidb-operator-overview.md:36]:
```suggestion
        - [在 AWS EKS 上部署 TiDB 集群](deploy-on-aws-eks.md)
```

File: zh/tidb-operator-overview.md [hunk 3]
```diff
@@ -34,7 +34,7 @@ TiDB Operator 提供了多种方式来部署 Kubernetes 上的 TiDB 集群：
...
-        - [Deploy TiDB on GCP GKE (beta)](deploy-on-gcp-gke.md)
+        - [Deploy TiDB on Google Cloud GKE (beta)](deploy-on-gcp-gke.md)
         - [Deploy TiDB on Azure AKS](deploy-on-azure-aks.md)
         - [Deploy TiDB on Alibaba Cloud ACK](deploy-on-alibaba-cloud.md)
```
Comment (Oreoxmt) [inline zh/tidb-operator-overview.md:39]:
```suggestion
        - [在 Google Cloud GKE 上部署 TiDB 集群 (beta)](deploy-on-gcp-gke.md)
        - [在 Azure AKS 上部署 TiDB 集群](deploy-on-azure-aks.md)
        - [在阿里云 ACK 上部署 TiDB 集群](deploy-on-alibaba-cloud.md)
```

@csuzhangxc Please confirm whether `deploy-on-gcp-gke.md` is currently in beta. If it is in beta, we need to make the following updates:
- update the title and add a note in https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/deploy-on-gcp-gke
- update the link text when referring to `deploy-on-gcp-gke.md`, such as:

https://github.com/pingcap/docs-tidb-operator/blob/63ee2f97824944cbc80a6cfe6b7ce3704037f293/zh/get-started.md?plain=1#L843-L846

Comment (csuzhangxc) [reply zh/tidb-operator-overview.md:39]: I think we can remove `beta` now.

File: en/tidb-operator-overview.md [hunk 4]
```diff
@@ -31,7 +31,7 @@ TiDB Operator provides several ways to deploy TiDB clusters on Kubernetes:
 
     + On public cloud:
         - [Deploy TiDB on AWS EKS](deploy-on-aws-eks.md)
-        - [Deploy TiDB on GCP GKE (beta)](deploy-on-gcp-gke.md)
+        - [Deploy TiDB on Google Cloud GKE (beta)](deploy-on-gcp-gke.md)
```
Comment (lilin90) [inline en/tidb-operator-overview.md:34]:
```suggestion
        - [Deploy TiDB on Google Cloud GKE](deploy-on-gcp-gke.md)
```

File: zh/tidb-operator-overview.md [hunk 5]
```diff
@@ -33,10 +33,10 @@ TiDB Operator 提供了多种方式来部署 Kubernetes 上的 TiDB 集群：
 
     - 在公有云上部署生产可用的 TiDB 集群并进行后续的运维管理；
 
-        - [Deploy TiDB on AWS EKS](deploy-on-aws-eks.md)
-        - [Deploy TiDB on GCP GKE (beta)](deploy-on-gcp-gke.md)
-        - [Deploy TiDB on Azure AKS](deploy-on-azure-aks.md)
-        - [Deploy TiDB on Alibaba Cloud ACK](deploy-on-alibaba-cloud.md)
+        - [在 AWS EKS 上部署 TiDB 集群](deploy-on-aws-eks.md)
+        - [在 Google Cloud GKE 上部署 TiDB 集群 (beta)](deploy-on-gcp-gke.md)
```
Comment (lilin90) [inline zh/tidb-operator-overview.md:37]:
```suggestion
        - [在 Google Cloud GKE 上部署 TiDB 集群](deploy-on-gcp-gke.md)
```

**Comment by lilin90** [comment]
Removed "(beta)" from GKE as confirmed by @csuzhangxc in https://github.com/pingcap/docs-tidb-operator/pull/2368#discussion_r1205044728 @Oreoxmt PTAL again~ Thanks!

**Review by csuzhangxc — APPROVED** [review]

**Review by csuzhangxc — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Pull request merged**